doc/api/n-api.md

```diff
@@ -1585,23 +1616,30 @@ added: REPLACEME
 ```c
 NAPI_EXTERN napi_status napi_add_async_cleanup_hook(
     napi_env env,
-    void (*fun)(void* arg, void(* cb)(void*), void* cbarg),
+    napi_async_cleanup_hook hook,
```

Comment:
If you're removing the signature here, please add a link to the definition of `napi_async_cleanup_hook`

Reply:
Added the link [below](https://github.com/nodejs/node/pull/34819/files#diff-6782512949f60a32853d80270ceefeb0R5812).

Reply:
I do like how this better hides the internal implementation. It seemed like we were leaking a bit more than needed in the N-API APIs wrapping the internal node api.